kratos/utilities/element_size_calculator.cpp

```diff
@@ -190,49 +289,317 @@ double ElementSizeCalculator<2,4>::MinimumElementSizeDerivative(
     const double y21 = (r_node_2.Y() + r_node_1.Y())/2.;
     const double y21_derivative = ((DerivativeNodeIndex == 2) + (DerivativeNodeIndex == 1)) * (DerivativeDirectionIndex == 1) / 2.;
 
-    const double x32 = (r_node_3.X() + r_node_2.X())/2.;
-    const double x32_derivative = ((DerivativeNodeIndex == 3) + (DerivativeNodeIndex == 2)) * (DerivativeDirectionIndex == 0) / 2.;
+    const double x32 = (r_node_3.X() + r_node_2.X())/2.;
+    const double x32_derivative = ((DerivativeNodeIndex == 3) + (DerivativeNodeIndex == 2)) * (DerivativeDirectionIndex == 0) / 2.;
+
+    const double y32 = (r_node_3.Y() + r_node_2.Y())/2.;
+    const double y32_derivative = ((DerivativeNodeIndex == 3) + (DerivativeNodeIndex == 2)) * (DerivativeDirectionIndex == 1) / 2.;
+
+    const double x03 = (r_node_0.X() + r_node_3.X())/2.;
+    const double x03_derivative = ((DerivativeNodeIndex == 0) + (DerivativeNodeIndex == 3)) * (DerivativeDirectionIndex == 0) / 2.;
+
+    const double y03 = (r_node_0.Y() + r_node_3.Y())/2.;
+    const double y03_derivative = ((DerivativeNodeIndex == 0) + (DerivativeNodeIndex == 3)) * (DerivativeDirectionIndex == 1) / 2.;
+
+    // Distance between face centers (xi direction)
+    const double dxi_x = x21 - x03;
+    const double dxi_x_derivative = x21_derivative - x03_derivative;
+
+    const double dxi_y = y21 - y03;
+    const double dxi_y_derivative = y21_derivative - y03_derivative;
+
+    const double h2_xi = dxi_x*dxi_x + dxi_y*dxi_y;
+    const double h2_xi_derivative = 2 * dxi_x * dxi_x_derivative + 2 * dxi_y * dxi_y_derivative;
+
+    // Distance between face centers (eta direction)
+    const double deta_x = x32 - x10;
+    const double deta_x_derivative = x32_derivative - x10_derivative;
+
+    const double deta_y = y32 - y10;
+    const double deta_y_derivative = y32_derivative - y10_derivative;
+
+    const double h2_eta = deta_x*deta_x + deta_y*deta_y;
+    const double h2_eta_derivative = 2 * deta_x * deta_x_derivative + 2 * deta_y * deta_y_derivative;
+
+    const double h2 = h2_xi < h2_eta ? h2_xi : h2_eta;
+    const double h2_derivative = h2_xi < h2_eta ? h2_xi_derivative : h2_eta_derivative;
+
+    return 0.5 * h2_derivative / std::sqrt(h2);
+
+    KRATOS_CATCH("");
+}
+
+// Quadrilateral2D9 version.
+template<>
+double ElementSizeCalculator<2,9>::MinimumElementSize(const Geometry<Node<3> >& rGeometry)
+{
+
```

Comment:
Same comment. This implementation is the same than that of the linear quadrilateral. Call the other method in here.